src/modalities/nn/weight_init/low_level_weight_init_factory.py

```diff
@@ -0,0 +1,121 @@
+import math
+from typing import Annotated, List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+from modalities.config.pydanctic_if_types import PydanticWeightInitializationIFType
+from modalities.nn.weight_init.weight_init import (
+    ModulewiseNormalInitialization,
+    NamedParameterwiseNormalInitialization,
+    WeightInitializationIF,
+    WeightInitializerWrapper,
+)
+
+
+class WeightInitializerWrapperConfig(BaseModel):
+    weight_initializers: List[PydanticWeightInitializationIFType]
+
+
+class PlainWeightInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    hidden_dim: Optional[int] = None
+
+    @model_validator(mode="after")
+    def check_std_and_hidden_dim(self):
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
+        return self
```

Comment:
`self.std` can be `auto` also for initialization types other than `plain`. So if we want to check this here (we seem to check it in 3 different places in total: `ComposedWeightInitializationConfig`, `PlainWeightInitializationConfig `, `LowLevelInitializationFactory`), we need to do it for `scaled` and `scaled_embed` as well.

Reply:
see [comment](https://github.com/Modalities/modalities/pull/168#discussion_r1663192270)